SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/Test.swift

```diff
@@ -0,0 +1,163 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+
+struct Test: ParsableCommand, BuildCommand {
+  @OptionGroup
+  var arguments: BuildArguments
+
+  @Flag(help: "Don't run lit-based tests")
+  var skipLitTests: Bool = false
+
+  @Option(
+    help: """
+      Path to the FileCheck executable that was built as part of the LLVM repository.
+      If not specified, it will be looked up from PATH.
+      """
+  )
+  var filecheckExec: String?
+
+  func run() throws {
+    try buildProduct(productName: "lit-test-helper")
+    try buildExample(exampleName: "ExamplePlugin")
+
+    try runTests()
+
+    logSection("All tests passed")
+  }
+
+  private func runTests() throws {
+    logSection("Running SwiftSyntax Tests")
+
+    if !skipLitTests {
+      try runLitTests()
+    }
+
+    try runXctests()
+  }
+
+  private func runLitTests() throws {
+    logSection("Running lit-based tests")
+
+    guard FileManager.default.fileExists(atPath: Paths.litExec.path) else {
+      throw ScriptExectutionError(
+        message: """
+          Error: Could not find lit.py.
+          Looking at '\(Paths.litExec.path)'.
+
+          Make sure you have the llvm repo checked out next to the swift-syntax repo.
+          Refer to README.md for more information.
+          """
+      )
+    }
+
+    let productBinPath = try findProductBinPath()
+    let examplesBinPath = try findExamplesBinPath()
+
+    let litTestHelperExec = productBinPath.appendingPathComponent("lit-test-helper")
+    var litCall = [
+      Paths.litExec.path,
+      Paths.packageDir.path,
+      "lit_tests",
+    ]
+
+    if let filecheckExec {
+      litCall += ["--param", "FILECHECK=" + filecheckExec]
+    }
+
+    litCall += ["--param", "LIT_TEST_HELPER=" + litTestHelperExec.path]
+
+    litCall += ["--param", "EXAMPLES_BIN_PATH=" + examplesBinPath.path]
+    litCall += ["--param", "TOOLCHAIN=" + arguments.toolchain.path]
+
+    // Print all failures
+    litCall += ["--verbose"]
+    // Don't show all commands if verbose is not enabled
+    if !arguments.verbose {
+      litCall += ["--succinct"]
+    }
+
+    guard let pythonExec = Paths.python3Exec else {
+      throw ScriptExectutionError(message: "Didn't find python3 executable")
+    }
+
+    let process = ProcessRunner(
+      executableURL: pythonExec,
+      arguments: litCall
+    )
+
+    try process.run()
+  }
+
+  private func runXctests() throws {
```

Comment:
Nit, I would capitalize `C` and `T`

```suggestion
  private func runXCTests() throws {
```